sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessEvaluationContext.java

```diff
@@ -359,6 +360,23 @@ public CounterSet getCounters() {
    * Returns true if all steps are done.
    */
   public boolean isDone() {
-    return watermarkManager.isDone();
+    boolean done = true;
+    if (!options.isShutdownUnboundedProducersWithMaxWatermark() && containsUnboundedPCollection()) {
+      return false;
+    }
+    done &= watermarkManager.allWatermarksAtPositiveInfinity();
```

Comment:
Style nit: might as well be parallel to the prior construction, aka

```
if (!watermarkManager.allWatermarksAtPositiveInfinity()) {
  return false;
}
```

Just noticed because the original value of `done` is unused.